### PR TITLE
Refactors related venues usage

### PR DESCRIPTION
**Pull request by mike-lvov**
Instead of using tree-like fetch of related venues, we utilize new `worldId` property of the venue which will be introduced by https://github.com/sparkletown/sparkle/pull/2295

Do not merge until https://github.com/sparkletown/sparkle/pull/2295 and the run of the script to assign world id to every venue

### CODE REVIEW

**Review by dmitrymarchuk — CHANGES_REQUESTED** [review]
Great work @mike-lvov!👍
Left some comments & requests.

File: src/utils/venue.ts [hunk 1]
```diff
@@ -165,3 +165,45 @@ export const sortVenues = (
...
+  previouslyCheckedVenueIds?: readonly string[];
+  maxDepth?: number;
+}
+
+export interface GetSovereignVenueReturn {
+  sovereignVenue: WithId<AnyVenue>;
+  checkedVenueIds: readonly string[];
+}
+
+export const getSovereignVenue = (
```
[the remark this answers is not in the document]
Comment (mike-lvov) [reply src/utils/venue.ts]: This function is used as an implementation detail in the `useRelatedVenues` hooks.

Renamed in https://github.com/sparkletown/sparkle/pull/2307/commits/4fdee3459a48ef2dbbe3df9e2dc433cc5dd2634e

File: src/hooks/useRelatedVenues.tsx [hunk 2]
```diff
@@ -94,34 +91,27 @@ export const RelatedVenuesProvider: React.FC<RelatedVenuesProviderProps> = ({
...
     () => ({
-      isLoading: isSovereignVenueLoading || isDescendantVenuesLoading,
-      isError: isTruthy(sovereignVenueError || descendantVenuesError),
+      isLoading: !isLoaded(relatedVenues),
 
-      isSovereignVenueLoading,
       sovereignVenue,
       sovereignVenueId,
-      sovereignVenueError,
+      checkedVenueIds,
```
[the remark this answers is not in the document]
Comment (mike-lvov) [reply src/hooks/useRelatedVenues.tsx]: Renamed in https://github.com/sparkletown/sparkle/pull/2307/commits/4fdee3459a48ef2dbbe3df9e2dc433cc5dd2634e

It's legacy logic & this PR is only to make it work with new connection type

File: src/hooks/useRelatedVenues.tsx [hunk 3]
```diff
@@ -154,10 +144,12 @@ export interface RelatedVenuesData extends RelatedVenuesContextState {
...
-export const useRelatedVenues: ReactHook<
-  RelatedVenuesProps,
-  RelatedVenuesData
-> = ({ currentVenueId }): RelatedVenuesData => {
+export function useRelatedVenues(props: RelatedVenuesProps): RelatedVenuesData;
+export function useRelatedVenues(): RelatedVenuesContextState;
+
+// eslint-disable-next-line func-style, prefer-arrow/prefer-arrow-functions
+export function useRelatedVenues(props?: RelatedVenuesProps) {
+  const { currentVenueId } = props ?? {};
```
[the remark this answers is not in the document]
Comment (mike-lvov) [reply src/hooks/useRelatedVenues.tsx:153]: there is already a separate useCurrentVenue(even 2 of them). The issue is that there are a lot of places where this hook is used to find a venue by its current venueId

File: src/api/venue.ts [hunk 4]
```diff
@@ -54,42 +50,6 @@ export interface FetchSovereignVenueOptions {
...
-    throw new Error("Maximum depth reached before finding the sovereignVenue.");
-
-  return fetchSovereignVenue(venue.parentId, {
-    ...options,
-    previouslyCheckedVenueIds: [...previouslyCheckedVenueIds, venueId],
-    maxDepth: maxDepth ? maxDepth - 1 : undefined,
-  });
-};
-
 export const fetchVenue = async (
```
[the remark this answers is not in the document]
Comment (mike-lvov) [reply src/api/venue.ts]: Deleted in https://github.com/sparkletown/sparkle/pull/2307/commits/4fdee3459a48ef2dbbe3df9e2dc433cc5dd2634e

File: src/components/molecules/ScheduleVenueDescription/ScheduleVenueDescription.tsx [hunk 5]
```diff
@@ -11,16 +11,10 @@ import { RenderMarkdown } from "components/organisms/RenderMarkdown";
...
-  venueId: string;
-}
-
-export const ScheduleVenueDescription: FC<ScheduleVenueDescriptionProps> = ({
-  venueId,
-}) => {
-  const { sovereignVenue } = useRelatedVenues({
-    currentVenueId: venueId,
-  });
+export interface ScheduleVenueDescriptionProps {}
```
[the remark this answers is not in the document]
Comment (mike-lvov) [reply src/components/molecules/ScheduleVenueDescription/ScheduleVenueDescription.tsx]: Removed in https://github.com/sparkletown/sparkle/pull/2307/commits/4fdee3459a48ef2dbbe3df9e2dc433cc5dd2634e

File: src/utils/venue.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmitrymarchuk) [reply src/utils/venue.ts]: But it's in `utils` and is `export`ed so one might assume it's a global utility function.
You might move it to`useRelatedVenues` file and do not `export` it there.

**Review by dmitrymarchuk — APPROVED** [review]
LGTM

**Pull request merged**